stlab/forest.hpp

```diff
@@ -546,6 +546,8 @@ struct forest_iterator {
 
     forest_iterator(const forest_iterator& x) : _node(x._node), _edge(x._edge) {}
 
+    forest_iterator& operator=(const forest_iterator&) = default;
```

Comment:
See here comment on the Slack channel.

Reply:
I transcribe here what was written in Slack:

FelixPetriconi said:

> @foster I just had a look into PR #333 and there is a change request for the "forest_iterator". I am asking why there is an explicit copy'ctor any case. Wouldn't the default for all c'tors and assignment operators be OK?

fpelliccioni said:

> I think the explicit copy ctor is not necessary, but...

fosterbrereton said:

> I do not think it needs to be explicit, either.

fpelliccioni said:

> I did not want to fix it in this PR, but GCC 10 gave me a warning (and -Werror) so I had the fix it in this way.
> It is not the purpose of the PR, so I did not want to analyze the code in detail.
> I propose that we create an issue to solve it in another PR.
> What do you think?

fosterbrereton said:

> That works for me. What’s the GCC warning? (I’m curious why clang doesn’t emit the same?)

fpelliccioni said:

> Compiler generation of copy assignment operator is deprecated (I do not remember in which C++ Standard), because of the explicit copy constructor.
> 
> https://github.com/fpelliccioni/libraries/runs/1545675743?check_suite_focus=true#step:7:1083

Reply:
I created these 2 related issues:

https://github.com/stlab/libraries/issues/334
https://github.com/stlab/libraries/issues/335